sdks/python/apache_beam/io/gcp/datastore/v1new/util.py

```diff
@@ -137,3 +137,22 @@ def report_latency(self, now, latency_ms, num_mutations):
       num_mutations: int, number of mutations contained in the RPC.
     """
     self._commit_time_per_entity_ms.add(now, latency_ms / num_mutations)
+
+
+def extract_byte_size(proto_message):
+  """
+    Gets the byte size from a google.protobuf or proto-plus message
```

Comment:
Let's increase the lower bound on Datastore and assume proto-plus messages for simplicity. the v2 dependency has been available for more than 1 year now, I don't think we need to support v1.
cc: @johnjcasey FYI.

Reply:
I don't know of any reason to not stop supporting v1

Reply:
There are some helper functions which converts key or entity to protobuf. There are certain piece of test code that replicates this behavior using `FakeMutation` object. In tests, FakeMutation object has attr ByteSize() in which we calculate byte size of messages. 

To replace the FakeMutation class, we need to refactor some pieces of code. So for now, I am keeping this as it is  now and I filed an issue https://github.com/apache/beam/issues/25625 to tackle this not only for this library but for other GCP dependencies as well.

Reply:
- you can access ByteSize in ProtoPlus message by msg._pb.ByteSize(). We access the ._pb in other places in Beam already.
- you can modify FakeMutation to support the accessor as above.  
-  given that we increase 'google-cloud-datastore>=2.0.0,<3',  I don't see a reason to support older version of the messages.  FakeMutation is purely test code, so keeping branching in the SDK code to accommodate the test code sounds like unnecessary complexity.